controllers/util/util.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetConditionByType returns condition of type T if it exists in conditions, otherwise
+// return empty condition struct.
+func GetConditionByType(t string, conditions []metav1.Condition) metav1.Condition {
+	for _, c := range conditions {
+		if c.Type == t {
+			return c
+		}
+	}
+	return metav1.Condition{}
+}
+
+func GetDeploymentCondition(status appsv1.DeploymentStatus, condType appsv1.DeploymentConditionType) *appsv1.DeploymentCondition {
+	for i := range status.Conditions {
```

Comment:
it feels like you should be able to simply this to
`for i, c := range status.Conditions {`, and even then `i` is unnecessary and could be `_`